vowpalwabbit/distributionally_robust.h

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright (c) by respective owners including Yahoo!, Microsoft, and
+individual contributors. All rights reserved.  Released under a BSD
+license as described in the file LICENSE.
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <limits>
+#include <stdexcept>
+
+namespace VW
+{
+namespace distributionally_robust
+{
+
+class ChiSquared
```

Comment:
This entire class could likely be marked constexpr, but probably not worth the effort at the moment :)

Reply:
Not done.